bob/src/cluster/quorum.rs

```diff
@@ -77,6 +78,50 @@ impl Quorum {
         }
     }
 
+    async fn delete_at_least(&self, key: BobKey, with_aliens: bool) -> Result<(), Error> {
+        debug!("DELETE[{}] ~~~DELETE LOCAL NODE FIRST~~~", key);
+        let mut local_put_ok = 0_usize;
+        let mut remote_ok_count = 0_usize;
+        let mut at_least = self.quorum;
+        let mut failed_nodes = Vec::new();
+        let res = delete_at_local_node(&self.backend, key, with_aliens).await;
+        if let Err(e) = res {
+            error!("{}", e);
+            failed_nodes.push(self.mapper.local_node_name().to_owned());
+        } else {
+            local_put_ok += 1;
+            at_least -= 1;
+            debug!("DELETE[{}] local node delete successful", key);
+        }
+        debug!(
+            "DELETE[{}] need at least {} additional deletes",
+            key, at_least
+        );
+
+        debug!("DELETE[{}] ~~~DELETE TO REMOTE NODES~~~", key);
+        let (tasks, errors) = self
+            .delete_at_remote_nodes(key, with_aliens, at_least)
+            .await;
+        let all_count = self.mapper.get_target_nodes_for_key(key).len();
+        remote_ok_count += all_count - errors.len() - tasks.len() - local_put_ok;
+        failed_nodes.extend(errors.iter().map(|e| e.node_name().to_string()));
+        if remote_ok_count + local_put_ok >= self.quorum {
+            debug!("DELETE[{}] spawn {} background put tasks", key, tasks.len());
+            let q = self.clone();
+            tokio::spawn(q.background_delete(tasks, key, failed_nodes));
```

Comment:
`delete` can be considered successful only if it successful on every node, thus running this operation in background is incorrect